src/DiscGolfScorecard1.java

```diff
@@ -0,0 +1,365 @@
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+/**
+ * {@code DiscGolfScorecard} represented as an array with implementations of
+ * primary methods.
+ *
+ * @convention 0 < $this.currentHoleNumber <= |$this.rep| and 0 <
+ *             $this.startingHoleNumber <= |$this.rep| and [all holes in
+ *             $this.rep have entries >= 0]
+ * @correspondence this = [holes in this.rep with initial starting hole
+ *                 $this.startingHoleNumber and current hole
+ *                 $this.currentHoleNumber]
+ *
+ * @author Luke Zhang
+ */
+public class DiscGolfScorecard1 extends DiscGolfScorecardSecondary {
+
+    /*
+     * Private members ---------------------------------------------------------
+     */
+
+    /**
+     * Default length of scorecard.
+     */
+    private static final int DEFAULT_SCORECARD_LENGTH = 18;
```

Comment:
You might place these constants in the kernel interface if these are to be shared by the whole package.